ref(storybook): add primitives section

JonasBa opened the pull request:
Adds a new 'primitives' category to the storybook navigation tree to group
fundamental layout and text components separately from other core components.

- Add 'primitives' to StoryCategory type
- Update category ordering to place primitives after foundations
- Create isPrimitivesFile() to identify core/layout and core/text components
- Modify inferFileCategory() to categorize primitives components correctly

This improves component organization by creating a dedicated section for
the most basic UI building blocks.

JonasBa committed:
feat(storybook): add primitives category for core layout and text components

Adds a new 'primitives' category to the storybook navigation tree to group
fundamental layout and text components separately from other core components.

- Add 'primitives' to StoryCategory type
- Update category ordering to place primitives after foundations
- Create isPrimitivesFile() to identify core/layout and core/text components
- Modify inferFileCategory() to categorize primitives components correctly

This improves component organization by creating a dedicated section for
the most basic UI building blocks.

JonasBa committed:
ref(stories): add sidebar primitives entry

TkDodo reviewed (commented):
what’s our final grouping that we want to have? I don’t think it should be `core` and `primitives`, but rather something like:

- `layout`
- `typography`
...

I also like the grouping that react aria has:

<img width="250" height="476" alt="Image" src="https://github.com/user-attachments/assets/50241e3b-3d3b-4ad9-985a-ae05ccf7608f" />

natemoo-re commented:
Started a conversation in Slack! I also really like React ARIA's grouping. I think we might want to use subsections for these, which might require a bit of a refactor? The `StoryTree` already supports collapsible sections, so this could use that system to become `Core > Primitives > Layout` and eventually things like  `Core > Controls > Button` and `Core > Overlays > Tooltip`

JonasBa commented:
Let me hold off on this until Grid and Container docs are merged, and I'll followup with changing this to Layout and Typography sections

JonasBa committed:
Merge branch 'master' into jb/storybook/primitives

JonasBa committed:
storybook: decouple text and heading docs

JonasBa commented:
@nate, I've updated the docs so that Text and Heading have separate docs, and we split layout from typography

<img width="996" height="1326" alt="CleanShot 2025-07-28 at 18 06 30@2x" src="https://github.com/user-attachments/assets/f92b7f2d-19c6-45e4-a135-1b564cf15aea" />

JonasBa committed:
heading: exclude ellipsis

JonasBa committed:
heading: remove unused imports

JonasBa committed:
Merge branch 'master' into jb/storybook/primitives

JonasBa committed:
text: remove heading section from text.mdx

JonasBa committed:
add back color omit

natemoo-re reviewed (approved):
Looks great! The only thing missing is keeping `core` as a section in `storySearch`, but I think we could do that in a follow-up. I'd like to refactor the categorization to remove the **`useStoryBookFilesByCategory`** function entirely and go back to having everything under `StoryCategory`.

JonasBa commented:
> Looks great! The only thing missing is keeping `core` as a section in `storySearch`, but I think we could do that in a follow-up. I'd like to refactor the categorization to remove the **`useStoryBookFilesByCategory`** function entirely and go back to having everything under `StoryCategory`.

Oh yes, let me add that so that those stories can be properly searched!

JonasBa committed:
stories: add typography and layout to search

JonasBa committed:
reorder search
